SConstruct

```diff
@@ -655,6 +655,22 @@ print(f'Building for platform "{platform_string}", architecture "{env["arch"]}",
 if env.dev_build:
     print_info("Developer build, with debug optimization level and debug symbols (unless overridden).")
 
+# Set x86 CPU instruction sets to use by the compiler's autovectorization.
+if env["arch"] == "x86_64" or env["arch"] == "x86_32":
+    if env.msvc and not methods.using_clang(env) and env["arch"] == "x86_64":
+        # https://stackoverflow.com/questions/64053597/how-do-i-enable-sse4-1-and-sse3-but-not-avx-in-msvc/69328426#69328426
+        env.Append(CCFLAGS=["/d2archSSE42"])
+    else:
+        # Be more conservative with instruction sets on 32-bit x86 to improve compatibility.
+        # SSE and SSE2 are present on all CPUs that support 64-bit, even if running a 32-bit OS.
+        env.Append(CCFLAGS=["-msse", "-msse2", "-mxsave"])
```

Comment:
This should only be set for `x86_32`, not `x86_64`. `-msse` should be implied by `-msse2` already.

What does `-mxsave` do?